cupy/cuda/memory.pyx

```diff
@@ -413,6 +413,12 @@ cdef class MemoryPointer:
             src (cupy.cuda.MemoryPointer): Source memory pointer.
             size (int): Size of the sequence in bytes.
 
+        .. warning::
+
+            This function always uses the legacy default stream and does not
+            honor the current stream. Use `copy_from_device_async` instead
+            if you don't understand what you are doing.
```

Comment:
I think that `copy_from_device_async` requires you to know what you are doing 😃.
Maybe we can change it to something in the lines of
`Use copy_from_device_async instead if you are using streams in your code, or have PTDS enabled`

Reply:
Right, updated as suggested! 👍🏼